waf

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env -S uv run --script
```

Comment:
same here, this creates a hidden development flow dependency on `uv`, which I'd avoid. If we use `uv`, let's make it transparent, explicit and optional.

Reply:
It's not really a hidden dependency, I've documented the requirement in the getting-started docs. I can drop this, but it means you'd have to prefix every Python command with `uv run` (worse for the docs - more like `SPHINXBUILD="uv run sphinx-build" make` which is *very* unintuitive). Feels like putting a fair bit of overhead on developers, which is part of what I was trying to reduce with this PR.

Reply:
the opposite version of this is: if you don't want to use `uv`, you can still `python3 waf build`

Reply:
By hidden dependency I mean `waf` now depends on `uv` to run, you have no other choice. Considering `uv` is not the official Python package manager solution, I'd avoid this. We should be able to still use `pip`.

Reply:
I'd agree it's hidden in a sense, but as @jwise pointed out you do have another choice.

This is also to my mind the biggest benefit of uv for me, so I'd prefer to keep it as it is currently in this PR.

Reply:
I'm not sold this is an advantage. If activating the venv is an issue, there are quite a few shell scripts/plugins etc that will automatically load venvs if found when entering a directory, and then, regardless of who created such venv or which tool, commands will work as usual.

Reply:
It doesn't - as Joshua mentioned, you can still use `python3 waf` and it will be run with whatever Python is currently in PATH, without interacting with uv at all.